srv/modules/utils/ready.py

```diff
@@ -53,20 +53,27 @@ def firewall(self):
         Scan all minions for the default firewall settings.  Set warnings
         for any differences.
         """
-        contents = self.local.cmd(self.search, 'cmd.shell',
+        contents = self.local.cmd(self.search, 'cmd.run_all',
                                   ['/usr/sbin/iptables -S'],
                                   tgt_type="compound")
+
         for minion in contents:
+            if contents[minion]['retcode'] == 127:
+                self.passed['firewall'] = "not installed"
+                continue
+            if 'firewall' in self.passed:
+                del self.passed['firewall']
```

Comment:
I'm confused -- doesn't this delete the value that was set in the first iteration of this loop?

Reply:
Yeah, I was trying to do the minimal impact to the current behavior.  In other words, the only way passed is set to "not installed" is if all minions hit the continue.  Then the condition to set passed to "disabled" is not triggered since it's already set.

If any minion has iptables return some other code other than 127, then all bets are off.  Delete the key and see if a warning is triggered.  If not, then the official passed is "disabled".  Another way to think about this is that "not installed" is a bad response if at least one minion does have iptables installed.

Reply:
Ah! Now I understand, thank you :-) Is it worth adding a comment to that effect?